arangod/Replication2/StateMachines/Document/DocumentFollowerState.cpp

```diff
@@ -45,7 +50,53 @@ auto DocumentFollowerState::acquireSnapshot(ParticipantId const& destination,
 
 auto DocumentFollowerState::applyEntries(
     std::unique_ptr<EntryIterator> ptr) noexcept -> futures::Future<Result> {
+  while (auto entry = ptr->next()) {
+    auto doc = entry->second;
+
+    auto transactionHandler = _guardedData.doUnderLock(
+        [](auto& data) -> std::shared_ptr<IDocumentStateTransactionHandler> {
+          if (data.didResign()) {
+            return nullptr;
+          }
+          return data.core->getTransactionHandler();
+        });
+    if (transactionHandler == nullptr) {
+      return {TRI_ERROR_CLUSTER_NOT_FOLLOWER};
+    }
+
+    try {
+      auto fut = futures::Future<Result>{Result{}};
+      auto trx = transactionHandler->ensureTransaction(doc);
+      TRI_ASSERT(trx != nullptr);
+      switch (doc.operation) {
+        case OperationType::kInsert:
+        case OperationType::kUpdate:
+        case OperationType::kReplace:
+        case OperationType::kRemove:
+        case OperationType::kTruncate:
+          fut = trx->apply(doc);
+          break;
+        case OperationType::kCommit:
+        case OperationType::kAbort:
+          fut = trx->finish(doc);
```

Comment:
How does that work? How do you distringuish between commit and abort?

Reply:
We look at the operation type inside the finish function. There's a case in which we have to abort anyways, even if the operation is commit. I am going to refactor this now to include a bug fix.